modAionImpl/src/org/aion/zero/impl/blockchain/AionHub.java

```diff
@@ -117,7 +117,7 @@ private void initializeHub(
         this.blockchain = _blockchain == null ? new AionBlockchainImpl(cfg, forTest) : _blockchain;
         blockchain.setEventManager(this.eventMgr);
 
-        this.mempool = AionPendingStateImpl.create(cfg, blockchain, forTest);
+        this.mempool = forTest ? AionPendingStateImpl.createForTest(cfg, blockchain) : AionPendingStateImpl.create(cfg, blockchain);
```

Comment:
I don't like this change. We should be increasing the functionality that can be tested not separating and reducing it further.